### PR TITLE
docs: add "Features" section to README

### PR DESCRIPTION
**Pull request by helenakallekleiv**
As described in https://github.com/equinor/terraform-baseline/issues/164

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,20 @@
...
+
+- This module creates an Azure Key Vault with the following features:
+  - Sets `standard` SKU.
+  - Sets soft-delete retention period for `90` days, by default.
+  - Disables purge protection, by default. Once enabled it cannot be disabled afterwards.
+  - Option for retrieving secrets by other Azure services, is disabled by default.
+  - Allows setting a list of access policies, empty by default.
+  - Enables RBAC authorization, by default.
+  - Enables public network access, by default.
+  - Denies network ACL's, by default.
```
Comment (hknutsen) [inline README.md:19]:
```suggestion
  - Network ACLs denies all traffic by default.
```

File: README.md [hunk 2]
```diff
@@ -6,6 +6,20 @@
...
+  - Enables Azure services to bypass network ACL's by default.
+  - Enables audit logs for diagnostoc setting, and sends diagnostics to a Log Analytics Workspace.
```
Comment (hknutsen) [inline README.md:21]:
```suggestion
  - Sends audit logs to given Log Analytics workspace.
```

**Comment by helenakallekleiv** [comment]
Keeping this as draft for now, until we (@equinor/terraform-baseline-maintainers) can discuss what features of the module should be included in the overview.

**Comment by hknutsen** [comment]
Tried to keep it even more to the point, and follow a common pattern `<FEATURE_NAME> <DEFAULT_BEHAVIOR>`:

```markdown
- Soft-delete retention set to 90 days by default.
- Purge protection disabled by default.
- Role-based access control (RBAC) authorization enabled by default.
- Public network access denied by default.
- Audit logs sent to given Log Analytics workspace by default.
```

I might've removed **too** much though 🤷 

@helenakallekleiv Maybe something in between our suggestions?

**Comment by helenakallekleiv** [comment]
> Tried to keep it even more to the point, and follow a common pattern `<FEATURE_NAME> <DEFAULT_BEHAVIOR>`:
> 
> ```
> - Soft-delete retention set to 90 days by default.
> - Purge protection disabled by default.
> - Role-based access control (RBAC) authorization enabled by default.
> - Public network access denied by default.
> - Audit logs sent to given Log Analytics workspace by default.
> ```
> 
> I might've removed **too** much though 🤷
> 
> @helenakallekleiv Maybe something in between our suggestions?

I like the common pattern method! 

What I think we should discuss is what do we look for when creating these lists, as of now its a little unclear to me. 
In the first draft I listed more or less everything this module does by default, based on the `main.tf` and `variables.tf` files, which I can see will become a little overkill when writing a "Features" section for bigger modules. 

So my question to @equinor/terraform-baseline-maintainers is "When should an argument or variable be consideren a feature, important enough (?), to be listed in the overview?" *or something like that, more or less "What scope are we thinking?"*

**Comment by helenakallekleiv** [comment]
Based on the discussion we had in the meeting today, we got some points/scope to follow when writing the "Features" overview:

- We try to follow a common naming pattern while writing the overview, e.g. `<FEATURE_NAME> <DEFAULT_BEHAVIOR>`.
- If there are two or more alternative features representing the same thing, only list the feature which are currently enabled. 
  - **Example:** Both RBAC authorization and access policies are alternative features delivered by the key-vault module, but only RBAC authorization is currently enabled. Therefore we only list RBAC authorization in the overview.
- If two or more arguments/variables delieveres features with a common purpose, we should try to summarize them in a single point.
  - **Example:** The suggested point "Public network access denied by default." summarizes the following points which are based on availiable variables:
    - "Enables public network access, by default."
    - "Network ACLs denies all traffic by default."
    - "Enables Azure services to bypass network ACL's by default."

**Review by hknutsen — APPROVED** [review]
Looks good 😎

**Pull request merged**